unified_planning/model/scheduling/chronicle.py

```diff
@@ -0,0 +1,147 @@
+                self._parameters[n] = up.model.parameter.Parameter(
+                    n, t, self._environment
+                )
+        else:
+            for n, t in kwargs.items():
+                assert self._environment.type_manager.has_type(
+                    t
+                ), "type of parameter does not belong to the same environment of the action"
+                self._parameters[n] = up.model.parameter.Parameter(
+                    n, t, self._environment
+                )
+
+    def __repr__(self) -> str:
+        s = []
+        s.append(f"{self.name}")
+        first = True
+        for p in self.parameters:
+            if first:
+                s.append("(")
+                first = False
+            else:
+                s.append(", ")
+            s.append(str(p))
+        if not first:
+            s.append(")")
+        s.append(" {\n")
+        if len(self._constraints) > 0:
+            s.append("    constraints = [\n")
+            for c in self._constraints:
+                s.append(f"      {str(c)}\n")
+            s.append("    ]\n")
+        if len(self.conditions) > 0:
+            s.append("    conditions = [\n")
+            for i, cl in self.conditions.items():
+                s.append(f"      {str(i)}:\n")
+                for c in cl:
+                    s.append(f"        {str(c)}\n")
+            s.append("    ]\n")
+        if len(self.effects) > 0:
+            s.append("    effects = [\n")
+            for t, el in self.effects.items():
+                s.append(f"      {str(t)}:\n")
+                for e in el:
+                    s.append(f"        {str(e)}:\n")
+            s.append("    ]\n")
+        s.append("  }")
+        return "".join(s)
+
+    def __eq__(self, oth: object) -> bool:
+        if not isinstance(oth, Chronicle):
+            return False
+        if (
+            self._environment != oth._environment
+            or self._parameters != oth._parameters
+            or self._name != oth._name
+        ):
+            return False
+        if set(self._constraints) != set(oth._constraints):
+            return False
+        if not TimedCondsEffs.__eq__(self, oth):
+            return False
+        return True
+
+    def __hash__(self) -> int:
+        res = hash(self._name)
+        res += sum(map(hash, self._parameters.items()))
+        res += sum(map(hash, self._constraints))
+        res += TimedCondsEffs.__hash__(self)
+        return res
+
+    @property
+    def name(self) -> str:
+        """Returns the `Chronicle` `name`."""
+        return self._name
+
+    def clone(self):
+        new = Chronicle(self._name, _env=self._environment)
+        new._parameters = self._parameters.copy()
+        new._constraints = self._constraints.copy()
+        TimedCondsEffs._clone_to(self, new)
+        return new
+
+    def add_parameter(self, name: str, tpe: Type) -> Parameter:
```

Comment:
Yes, in an `Action` parameters are paramount to the definition of the action because an action is essentially a template from which ground actions and then ground action instances will be generated.

In this spectrum, an `Activity ` is closest to a ground action instance and thus rarely need parameters. In fact a parameter of an activity is just a global decision variable, defined in the scope of the activity.
Not having them centric in the constructor allows letting more space for common parameters (duration, due date, ...).

Reply:
Ok! A short comment explaining the semantic of Activity parameters could be useful.